pkg/cmd/server/start/start_kube_controller_manager.go

```diff
@@ -77,22 +67,22 @@ func newKubeControllerManager(kubeconfigFile, saPrivateKeyFile, saRootCAFile, po
 	}
 	cmdLineArgs["openshift-config"] = []string{openshiftConfigFile}
 
-	// resolve arguments
-	controllerManager := controlleroptions.NewCMServer()
-	if err := cmdflags.Resolve(cmdLineArgs, kubeControllerManagerAddFlags(controllerManager)); len(err) > 0 {
-		return nil, kerrors.NewAggregate(err)
+	args := []string{}
+	for key, value := range cmdLineArgs {
+		for _, token := range value {
+			args = append(args, fmt.Sprintf("--%s=%v", key, token))
+		}
 	}
-
-	return controllerManager, nil
+	return args
 }
 
 func runEmbeddedKubeControllerManager(kubeconfigFile, saPrivateKeyFile, saRootCAFile, podEvictionTimeout, openshiftConfigFile string, dynamicProvisioningEnabled bool) {
-	// TODO we need a real identity for this.  Right now it's just using the loopback connection like it used to.
-	controllerManager, err := newKubeControllerManager(kubeconfigFile, saPrivateKeyFile, saRootCAFile, podEvictionTimeout, openshiftConfigFile, dynamicProvisioningEnabled)
-	if err != nil {
-		glog.Fatal(err)
-	}
-	if err := controllerapp.Run(controllerManager); err != nil {
+	cmd := controllerapp.NewControllerManagerCommand()
+	args := computeKubeControllerManagerArgs(kubeconfigFile, saPrivateKeyFile, saRootCAFile, podEvictionTimeout, openshiftConfigFile, dynamicProvisioningEnabled)
+	if err := cmd.ParseFlags(args); err != nil {
 		glog.Fatal(err)
 	}
+	glog.Infof("`kube-controller-manager %v`", args)
+	cmd.Run(nil, nil)
+	panic(fmt.Sprintf("`kube-controller-manager %v` exited", args))
```

Comment:
Is this intentional here?

Reply:
> Is this intentional here?

Yes.  Because we can't re-init, we need to die. And if we loop, we'll want to know why